frontend/components/SetupPage/SetupYourAgent/ModiusAgentForm/ModiusAgentForm.tsx

```diff
@@ -19,47 +20,68 @@ import {
 } from '../../../AgentForms/common/formUtils';
 import { InvalidGeminiApiCredentials } from '../../../AgentForms/common/InvalidGeminiApiCredentials';
 import {
+  CoinGeckoApiKeyDesc,
   CoinGeckoApiKeyLabel,
+  CoinGeckoApiKeySubHeader,
+  GeminiApiKeyDesc,
   GeminiApiKeyLabel,
+  GeminiApiKeySubHeader,
+  TenderlyAccessTokenDesc,
   TenderlyAccessTokenLabel,
   TenderlyAccountSlugLabel,
+  TenderlyApiKeySubHeader,
   TenderlyProjectSlugLabel,
 } from '../../../AgentForms/common/labels';
+import { RenderForm } from '../useDisplayAgentForm';
 import {
   ModiusFieldValues,
   useModiusFormValidate,
 } from './useModiusFormValidate';
 
-const { Text } = Typography;
-
-const SetupHeader = () => (
-  <Text>
-    Set up your agent with access to a{' '}
-    <a target="_blank" href={TENDERLY_URL}>
-      Tenderly
-    </a>{' '}
-    project for simulating bridge and swap routes, and swap routes and provide a{' '}
-    <a target="_blank" href={COINGECKO_URL}>
-      CoinGecko API key
-    </a>{' '}
-    as a price source.
-  </Text>
-);
-
-type ModiusAgentFormProps = { serviceTemplate: ServiceTemplate };
-
-export const ModiusAgentForm = ({ serviceTemplate }: ModiusAgentFormProps) => {
+type ModiusAgentFormContentProps = {
+  serviceTemplate: ServiceTemplate;
+  currentStep: BabyDegenFormStep;
+  updateNextStep: () => void;
+};
+
+export const ModiusAgentFormContent = ({
+  serviceTemplate,
+  currentStep,
+  updateNextStep,
+}: ModiusAgentFormContentProps) => {
+  const [form] = Form.useForm<ModiusFieldValues>();
   const { goto } = useSetup();
   const { defaultStakingProgramId } = useStakingProgram();
 
-  const [form] = Form.useForm<ModiusFieldValues>();
   const [isSubmitting, setIsSubmitting] = useState(false);
   const {
     geminiApiKeyValidationStatus,
     submitButtonText,
     updateSubmitButtonText,
     validateForm,
-  } = useModiusFormValidate();
+  } = useModiusFormValidate('Finish Agent Configuration');
+
+  const isTenderlyStep = currentStep === BABYDEGEN_FORM_STEP.tenderly;
+  const isCoinGeckoStep = currentStep === BABYDEGEN_FORM_STEP.coingecko;
+  const isGeminiStep = currentStep === BABYDEGEN_FORM_STEP.gemini;
+
+  const handleContinue = useCallback(async () => {
+    try {
+      if (isTenderlyStep) {
+        await form.validateFields([
+          'tenderlyAccessToken',
+          'tenderlyAccountSlug',
+          'tenderlyProjectSlug',
+        ]);
+        updateNextStep();
+      } else if (isCoinGeckoStep) {
+        await form.validateFields(['coinGeckoApiKey']);
+        updateNextStep();
+      }
```

Comment:
just a thought, we could have had generic steps like 1,2,3 each with titles, fields to validate, etc. so it's like a configurable template for a form, so that we could easily change steps content, etc.

also could have there which fields to hide/show, which is prev/next step, etc.